cpp/include/cudf/detail/null_mask.hpp

```diff
@@ -114,7 +114,7 @@ rmm::device_buffer copy_bitmask(
  *
  * @param stream CUDA stream used for device memory operations and kernel launches
  */
-rmm::device_buffer bitmask_and(
+std::pair<rmm::device_buffer, size_type> bitmask_and(
```

Comment:
Hm, I suppose there are times where we want this to return the count of _set_ bits, and other times where we want the count of _unset_ bits. For instance, `valid_if` always returns the count of unset bits (i.e., the "null count). Obviously its easy enough to compute one from the other, but that could be annoying... 

Maybe here we return a `struct` with both?

```
struct bitmask {
   rmm::device_buffer mask;
   size_type set_bits;
   size_type unset_bits;
};
```

Reply:
Returning the count of both _set_ bits and _unset_ bits seems redundant since at least one variable out of three won't be used.

Reply:
@jrhemstad `bitmask_and` now returns a struct of triplet as you suggested. As far as I know, hash join is the only case (maybe semi-join also, not sure) demanding `valid_count` in postprocessing and all others will use `null_count`.